### PR TITLE
TTF composite flags

### PR DESCRIPTION
**Pull request by jenskutilek**
Extract composite flags according to spec proposal https://github.com/unified-font-object/ufo-spec/pull/170

(Supersedes #32)

### CODE REVIEW

File: Lib/extractor/formats/opentype.py [hunk 1]
```diff
@@ -164,33 +171,24 @@ def _byteCodeToTtxAssembly(program):
...
     for ci, c in enumerate(glyph.components):
         flags = {}
-        if c.flags & ROUND_XY_TO_GRID:
-            flags["round"] = True
-        if c.flags & USE_MY_METRICS:
-            flags["useMyMetrics"] = True
-        if c.flags & OVERLAP_COMPOUND:
-            flags["overlap"] = True
+        flags[TRUETYPE_ROUND_KEY] = bool(c.flags & ROUND_XY_TO_GRID)
+        flags[TRUETYPE_METRICS_KEY] = bool(c.flags & USE_MY_METRICS)
```
Comment (moyogo) [inline Lib/extractor/formats/opentype.py:181]: Would it be better to only define  the flags when their values are true?

**Comment by benkiel** [comment]
@jenskutilek @moyogo let me know when this is good to merge.

File: Lib/extractor/formats/opentype.py (hunk 1 above)
Comment (jenskutilek) [reply Lib/extractor/formats/opentype.py:181]: Hm, that would probably depend on how presence/absence of the flags is interpreted by authoring tools or at compile time. I probably was thinking along the lines of

- Flag is not present: Use heuristic or use authoring tool default value
- Flag is present: Use flag value

But I guess that is not specifed anywhere in the UFO spec ...?

So for anything TrueType-related I wanted to be on the safe side and always set the flags explicitly.

Comment (moyogo) [reply Lib/extractor/formats/opentype.py:181]: That makes sense. Thanks.

**Comment by moyogo** [comment]
Looks good to me.

**Pull request merged**